redisinsight/ui/src/pages/home/components/AddInstanceControls/AddInstanceControls.tsx

```diff
@@ -1,33 +1,53 @@
-import React from 'react'
+import React, { useContext, useEffect, useState } from 'react'
 import {
   EuiButton,
   EuiFlexGroup,
   EuiFlexItem,
-  EuiIcon,
   EuiSpacer,
   EuiText,
 } from '@elastic/eui'
-import cx from 'classnames'
+import { isEmpty } from 'lodash'
+import { useSelector } from 'react-redux'
 import { sendEventTelemetry, TelemetryEvent } from 'uiSrc/telemetry'
 import HelpLinksMenu from 'uiSrc/pages/home/components/HelpLinksMenu'
-import { HELP_LINKS } from 'uiSrc/pages/home/constants/help-links'
+import PromoLink from 'uiSrc/components/promo-link/PromoLink'
+import { ThemeContext } from 'uiSrc/contexts/themeContext'
+import { contentSelector } from 'uiSrc/slices/content/content-create-database'
+import { HELP_LINKS, IHelpGuide } from 'uiSrc/pages/home/constants/help-links'
+import { getPathToResource } from 'uiSrc/services/resourcesService'
+import { ContentCreateDBItem } from 'uiSrc/slices/interfaces/content'
 
 import styles from './styles.module.scss'
 
-const SHOW_FOR_XL = 'eui-showFor--xl'
-
-export enum Direction {
-  column = 'column',
-  row = 'row',
-}
-
-interface Props {
+export interface Props {
   onAddInstance: () => void;
-  direction: Direction;
+  direction: 'column' | 'row';
   welcomePage?: boolean;
 }
 
 const AddInstanceControls = ({ onAddInstance, direction, welcomePage = false }: Props) => {
+  const [promoData, setPromoData] = useState<ContentCreateDBItem>()
+  const [guides, setGuides] = useState<IHelpGuide[]>([])
+  const { loading, data } = useSelector(contentSelector)
+  const { theme } = useContext(ThemeContext)
+
+  useEffect(() => {
+    if (!loading && !isEmpty(data)) {
```

Comment:
use return early here, please

Reply:
where did you find `return` from the function?

Reply:
done